From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Tue, 7 Aug 2018 10:19:43 -0700
Subject: [PATCH] fileattrs: Don't scan libraries in glibc auto-search dirs for
 Provides

Those libraries are meant to be runtime replacements of the ones in the regular
$libdir, based on CPU features at runtime. If a packager decides to split them into
a separate sub-package, the dependency resolver shouldn't be deciding to install
that instead of the base package.

[2021-05-05]: Remove elfoptimized.attr. This will installed by clr-rpm-config
              instead to ease maintenance.

[2021-09-10]: Ported for rpm 4.17.0 update

[2021-10-08]: Squashed all patches to elf.attr into this one; exclude
              new /usr/share/clear/optimized-elf tree as well.

[2022-06-26]: Updated regexp to ignore /usr/lib64/glibc-hwcaps tree

Ported-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>
Ported-by: Patrick McCarty <patrick.mccarty@intel.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 fileattrs/elf.attr | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fileattrs/elf.attr b/fileattrs/elf.attr
index 5cb54bc..505f1be 100644
--- a/fileattrs/elf.attr
+++ b/fileattrs/elf.attr
@@ -1,3 +1,4 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires
 %__elf_magic		ELF (32|64)-bit.*$
+%__elf_exclude_path     ^(/usr/lib/debug|/usr/mingw/lib|/usr/share/clear/optimized-elf|/usr/x86_64-generic-linux/x86_64-w64-mingw32/lib|(/usr)?/lib(64)?/(i686|x86_64|sse2|haswell|tls|openmpi|glibc-hwcaps))/
